lib/frames/transfer_frame.js

```diff
@@ -247,4 +255,66 @@ TransferFrame.prototype._getAdditionalPayload = function() {
   return this.message;
 };
 
+/**
+ * An AMQP Message is composed of:
+ *
+ * * Zero or one header
+ * * Zero or one delivery-annotations
+ * * Zero or one message-annotations
+ * * Zero or one properties
+ * * Zero or one application-properties
+ * * Body: One or more data sections, one or more amqp-sequence sections, or one amqp-value section
+ * * Zero or one footer
+ *
+ * @return {Message}               Complete message object decoded from buffer
+ */
+TransferFrame.prototype.decodePayload = function() {
+  var message = new M.Message();
+  var body = [];
+  var curIdx = 0;
+  var possibleFields = {
+    'header': M.Header, 'footer': M.Footer, 'deliveryAnnotations': M.DeliveryAnnotations,
+    'annotations': M.Annotations, 'properties': M.Properties, 'applicationProperties': M.ApplicationProperties
+  };
+  var isData = function(x) { return x instanceof M.Data; };
```

Comment:
not that I think its relevant for this PR, but I think these should be defined outside the function in the module, otherwise we're recreating this on _every_ `decodePayload` call